scalableminds/webknossos
# Revert "Revert "Add Recommended Configuration to Task Types""

Reverts scalableminds/webknossos#3502

This PR re-adds the recommended settings PR together with the authentication fix for setting the `lastTaskTypeId`.
It also addresses feedback from discuss https://discuss.webknossos.org/t/settings-via-task-type/1173/9?u=dwerner and adds the display names (and some comments) to the recommended settings creation view and the modal the users see.

### Steps to test:
- Login as a non-admin user and open the task. The `lastTaskTypeId` should be updated without throwing an authentication error alert.
- The Modal should only show iff the taskType changes.
- The meaning of the different settings should be clear to non-wK devs :)

![rec-config](https://user-images.githubusercontent.com/1702075/49378398-cb765c00-f70c-11e8-87f7-188832e62b15.png)
![rec-config-tracing-view](https://user-images.githubusercontent.com/1702075/49378405-ce714c80-f70c-11e8-8c1b-132e6ab3bc8d.png)
